corehq/motech/fhir/models.py

```diff
@@ -329,3 +329,100 @@ class Meta:
             models.Index(fields=['domain']),
             models.Index(fields=['frequency']),
         ]
+
+
+class FHIRImporterResourceType(models.Model):
+    """
+    Similar to ``FHIRResourceType`` but allows importing different case
+    types, with different mappings, from those used by ``FHIRRepeater``
+    and the FHIR API.
+
+    ``import_related_only`` determines whether the ``FHIRImporter``
+    searches the remote service for instances, or only imports instances
+    related to the resource types it is interested in. e.g. Only import
+    the Patients of imported ServiceRequests, not all Patients. Or only
+    import the Person resources who are the contacts of imported
+    Patients, not all Person resources.
+
+    If ``import_related_only`` is ``False`` (the default) , then
+    ``FHIRImporterResourceType`` imports resources filtered by the
+    ``search_params`` field.
+
+    .. note::
+
+       If ``import_related_only`` is ``False`` and ``search_params`` is
+       empty, then all resources of that type will be imported.
+
+    See the ``jsonpath_to_related_resource_type`` related name defined
+    in ``JSONPathToResourceType`` for details of how related resource
+    types are set.
+    """
+    fhir_importer = models.ForeignKey(
+        FHIRImporter,
+        on_delete=models.CASCADE,
+        related_name='resource_types',
+    )
+    name = models.CharField(max_length=255)
+    case_type = models.ForeignKey(CaseType, on_delete=models.CASCADE)
+
+    import_related_only = models.BooleanField(default=False)
+    search_params = JSONField(default=dict)
+
+
+class JSONPathToResourceType(models.Model):
```

Comment:
food-for-thought: Would something like "ResourceTypeRelationship" be a better name for this?  It seems like it's an intermediary model linking two resource types, and the JSON path is just a characteristic of that relationship.  `JSONPathToResourceType` sounds to me like a function that can evaluate a jsonpath to return a resource type.

Reply:
I struggled to come up with a name for this. "ResourceTypeRelationship" is a much better idea.